Tests/kaas/plugin/plugin_cluster_stacks.py

```diff
@@ -0,0 +1,327 @@
+
+        wait_for_pods(self, ["cso-system"], kubeconfig=self.kubeconfig_mgmnt)
+
+        # Create Cluster Stack definition and workload cluster
+        self._apply_yaml_with_envsubst("clusterstack.yaml", "Error applying clusterstack.yaml", kubeconfig=self.kubeconfig_mgmnt)
+        self._apply_yaml_with_envsubst("cluster.yaml", "Error applying cluster.yaml", kubeconfig=self.kubeconfig_mgmnt)
+
+        # Get and wait on kubeadmcontrolplane and retrieve workload cluster kubeconfig
+        kcp_name = self._get_kubeadm_control_plane_name(namespace=self.cs_namespace, kubeconfig=self.kubeconfig_mgmnt)
+        self._wait_kcp_ready(kcp_name, namespace=self.cs_namespace, kubeconfig=self.kubeconfig_mgmnt)
+        self._retrieve_kubeconfig(namespace=self.cs_namespace, kubeconfig=self.kubeconfig_mgmnt)
+
+        # Wait for workload system pods to be ready
+        # wait_for_workload_pods_ready(kubeconfig_path=self.kubeconfig_cs_cluster)
+        wait_for_pods(self, ["kube-system"], timeout=600, interval=15, kubeconfig=self.kubeconfig_cs_cluster)
+
+    def delete_cluster(self, cluster_name=None, kubeconfig_filepath=None):
+        self.cluster_name = cluster_name
+        kubeconfig_cs_cluster_filename = kubeconfig_filepath
+
+        # Get kubeconfig of the mgmnt (kind) cluster
+        self.cluster = KindCluster(cluster_name)
+        self.kubeconfig_mgmnt = str(self.cluster.kubeconfig_path.resolve())
+
+        try:
+            # Check if the cluster exists
+            check_cluster_command = f"kubectl get cluster {cluster_name} -n {self.cs_namespace}"
+            result = self._run_subprocess(check_cluster_command, "Failed to get cluster resource", shell=True, capture_output=True, text=True, kubeconfig=self.kubeconfig_mgmnt)
+
+            # Proceed with deletion only if the cluster exists
+            if result.returncode == 0:
+                delete_command = f"kubectl delete cluster {cluster_name} --timeout=600s -n {self.cs_namespace}"
+                self._run_subprocess(delete_command, "Timeout while deleting the cluster", shell=True, kubeconfig=self.kubeconfig_mgmnt)
+
+        except subprocess.CalledProcessError as error:
+            if "NotFound" in error.stderr:
+                logger.info(f"Cluster {cluster_name} not found. Skipping deletion.")
+            else:
+                raise RuntimeError(f"Error checking for cluster existence: {error}")
+
+        # Delete mgmngt (kind) cluster
+        self.cluster.delete()
+
+        # Remove kubeconfigs
+        if os.path.exists(kubeconfig_cs_cluster_filename):
+            os.remove(kubeconfig_cs_cluster_filename)
+        if os.path.exists(self.kubeconfig_mgmnt):
+            os.remove(self.kubeconfig_mgmnt)
+
+    def _apply_yaml_with_envsubst(self, yaml_file, error_msg, kubeconfig=None):
+        """
+        Applies a Kubernetes YAML configuration file to the cluster, substituting environment variables as needed.
```

Comment:
```suggestion
        Applies a Kubernetes YAML configuration file to the  mgmnt cluster, substituting environment variables as needed.
```
I would suggest referring to "mgmnt cluster" here to not confuse it with "workload clusters"